services/classification/luis/client_test.go

```diff
@@ -0,0 +1,88 @@
+package luis_test
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/nyaruka/goflow/services/classification/luis"
+	"github.com/nyaruka/goflow/utils/httpx"
+	"github.com/shopspring/decimal"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPredict(t *testing.T) {
+	httpx.SetRequestor(httpx.NewMockRequestor(map[string][]*http.Response{
+		"https://westus.api.cognitive.microsoft.com/luis/v2.0/apps/f96abf2f-3b53-4766-8ea6-09a655222a02/?verbose=true&subscription-key=3246231&q=Hello": []*http.Response{
+			httpx.NewMockResponse(200, `xx`), // non-JSON response
+			httpx.NewMockResponse(200, `{}`), // invalid JSON response
+			httpx.NewMockResponse(200, `{
+				"query": "book a flight to Quito",
+				"topScoringIntent": {
+				  "intent": "Book Flight",
+				  "score": 0.9106805
+				},
+				"intents": [
+				  {
+					"intent": "Book Flight",
+					"score": 0.9106805
+				  },
+				  {
+					"intent": "None",
+					"score": 0.08910245
+				  },
+				  {
+					"intent": "Book Hotel",
+					"score": 0.07790734
+				  }
+				],
+				"entities": [
+				  {
+					"entity": "quito",
+					"type": "City",
+					"startIndex": 17,
+					"endIndex": 21,
+					"score": 0.9644149
+				  }
+				],
+				"sentimentAnalysis": {
```

Comment:
should we just throw this in with the entities if it exists? it's a checkbox on the LUIS side

Reply:
Sure, so entity of type `sentiment` with the entity being the `label`, `score` etc..

Reply:
well our common entity format is `value` + `confidence` so it would just become one of those